homeassistant/components/ecobee/config_flow.py

```diff
@@ -0,0 +1,173 @@
+            """Config entry currently exists, only one allowed."""
+            return self.async_abort(reason="one_instance_only")
+
+        if user_input is not None:
+            """Use the user-supplied API key to attempt to obtain a PIN from ecobee."""
+            from pyecobee import Ecobee, ECOBEE_API_KEY
+
+            config = {ECOBEE_API_KEY: user_input[CONF_API_KEY]}
+
+            self._ecobee = Ecobee(config=config)
+
+            if await self.hass.async_add_executor_job(self._ecobee.request_pin):
+                """We have a PIN; move to the next step of the flow."""
+                return await self.async_step_authorize()
+            else:
+                """Obtaining the PIN failed. Maybe the wrong API key?"""
+                errors["base"] = "pin_request_failed"
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_API_KEY, default=stored_api_key): str}
+            ),
+            errors=errors,
+        )
+
+    async def async_step_authorize(self, user_input=None):
+        """Present the user with the PIN so that the app can be authorized on ecobee.com."""
+        errors = {}
+
+        if user_input is not None:
+            """Attempt to obtain tokens from ecobee and finish the flow."""
+            if await self.hass.async_add_executor_job(self._ecobee.request_tokens):
+                """Refresh token obtained; create the config entry."""
+                config = {
+                    CONF_API_KEY: self._ecobee.api_key,
+                    CONF_REFRESH_TOKEN: self._ecobee.refresh_token,
+                }
+                return self.async_create_entry(title=DOMAIN, data=config)
+            else:
+                errors["base"] = "token_request_failed"
+
+        return self.async_show_form(
+            step_id="authorize",
+            errors=errors,
+            description_placeholders={"pin": self._ecobee.pin},
+        )
+
+    async def async_step_import(self, import_data):
+        """Import ecobee config from an existing ecobee.conf file.
+
+        This flow is triggered by `async_setup` if no existing entry exists
+        and if ecobee.conf exists, allowing pre-config-flow ecobee component
+        users an easy migration path to using a config entry.
+
+        We will attempt to validate the credentials found in ecobee.conf
+        and create an entry if valid. Otherwise, we will abort and the user
+        will need to redo the authorization process via a new config flow.
+        """
+        import aiofiles
+        import json
+        from pyecobee import (
+            Ecobee,
+            ECOBEE_CONFIG_FILENAME,
+            ECOBEE_API_KEY,
+            ECOBEE_REFRESH_TOKEN,
+        )
+
+        config_file = self.hass.config.path(ECOBEE_CONFIG_FILENAME)
+
+        async with aiofiles.open(config_file, "r") as f:
+            legacy_config = json.loads(await f.read())
```

Comment:
We have `load_json` util to load json files. Please use that instead. Schedule it on the executor thread pool.

https://github.com/home-assistant/home-assistant/blob/1d3f2d20d2eb2d12d02cde93d7d79e66f8157a1c/homeassistant/util/json.py#L22